File: bubble_sort.cpp
#include <iostream>
#include <vector>

using std::vector;
using std::swap;
using std::cout;
using std::endl;

void bubbleSort(vector<int> nums, int size){
   int i;
   int j; 
   for (i = 0; i < size; ++i){
      for (j = 0; j < size - i; ++j){
         if (nums[j] > nums[j + 1])
            swap(nums[j], nums[j + 1]);
      }
   }
}

int main(){
   vector <int> someExample{12, 231, 25, 2551};

   bubbleSort(someExample, someExample.size());

   for (auto num : someExample)
      cout<<num<<endl;
}

File: bubble-sort.js


let someArray = [9, 8, 7, 6, 5, 4, 3, 2, 1];

function bubbleSort(nums){
	for (let i = 0; i < nums.length; i++) {
		for (let j = 0; j < nums.length - i - 1; j++) {
			if (nums[j] > nums[j+1]) {
				let temp = nums[j];
				nums[j] = nums[j + 1];
				nums[j + 1] = temp;
			}
		}
	}
};

console.log(someArray);
bubbleSort(someArray);
console.log(someArray);

